### ISSUE TITLE
Allow optionGroups/options to be sortable 

### ISSUE THREAD

**Issue by vmatekole** [vendure-ecommerce/vendure#865]
We have introduced variants to our products. Currently we are unable to sort the optionGroups or the options within them. This is a problem for when one wants to build dropdowns with intuitive order e.g. "Size" dropdown before "Colour".

Furthermore within the optionGroups you want to be able to order "Small" before "Large". As a workaround we have had to duplicate data of the optionGroups within the frontend, which is undesirable as it complicates product management and increases risk. GraphQl output below:

Example fo current graphQL output:
....
"optionGroups": [
        {
          "options": [
            {
              "group": {
                "name": "Size"
              },
              "name": "Large"
            },
            {
              "group": {
                "name": "Size"
              },
              "name": "Small"
            }
          ]
        },
        {
          "options": [
            {
              "group": {
                "name": "Wine"
              },
              "name": "+ with Wine"
            },
            {
              "group": {
                "name": "Wine"
              },
              "name": "- without Wine"
            }
          ]
        }
      ],
```

**Comment by peterjarian** [comment]
This has been an issue since 2021, and there is still no solution for this. It's really hard to advocate for Vendure as a good solution for full custom storefronts, if you don't have the option to sort how options will be shown on the backend. As a work-around I now need to keep the sorting on my frontend and that is just annoying.

**Comment by mzealey** [comment]
yes this is frustrating for me also - i assume we could add custom field of sortOrder or similar but it seems like this would best be done in the core + admin ui

**Comment by dlhck** [comment]
This feature request has been moved to GitHub Discussions for community voting.
Please upvote the discussion if you would like to see this implemented: https://github.com/vendurehq/vendure/discussions/4311
